File: Tutorial_Sample_Code/Other/CRUFL_ExchangeRate/VB/CRUFL_VB_ExchangeRate/ExchangeUfl.vb
Imports System.Runtime.InteropServices

<ComVisible(True), ClassInterface(ClassInterfaceType.None), GuidAttribute("CCF45F61-46D6-425A-AADF-2382774AE721")> _
Public Class ExchangeUfl : Implements IExchangeUfl
    Public Function ConvertUSDollarsToCDN1(ByVal usd As Double) As Double Implements IExchangeUfl.ConvertUSDollarsToCDN
        If usd > Double.MaxValue Then
            Throw New Exception("Value submitted is larger than the maximum value allowed for a double.")
        End If
        Return (usd * 1.45)
    End Function
End Class

File: Tutorial_Sample_Code/Other/CRUFL_ExchangeRate/CS/CRUFL_CS_ExchangeRate/ExchangeUfl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.InteropServices;


namespace CRUFL_CS_ExchangeRate
{
    [ComVisible(true), ClassInterface(ClassInterfaceType.None), Guid("E82DD73A-B76D-4C69-8C67-CD135C5111A3")]
    public class ExchangeUfl : IExchangeUfl
    {
        public double ConvertUSDollarsToCDN(double usd)
        {
            if (usd > Double.MaxValue)
            {
                throw new Exception("Value submitted is larger than the maximum value allowed for a double.");
            }
            return (usd * 1.45);
        }

    }


}
